### ISSUE TITLE
provider 3.7.0 - azurerm_private_endpoint - correct casing issue for private_connection_resource_id field for Microsoft.DBforPostgreSQL

### ISSUE THREAD

**Issue by igor-teresco** [hashicorp/terraform-provider-azurerm#17458]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.2.1

### AzureRM Provider Version

3.7.0

### Affected Resource(s)/Data Source(s)

azurerm_private_endpoint

### Terraform Configuration Files

```hcl
resource "azurerm_private_endpoint" "plink-pg-openvpn" {
  name                = "${var.environment}-psql-openvpn-endpoint"
  location            = var.location
  resource_group_name = azurerm_resource_group.kube.name
  subnet_id           = module.postgres-openvpn-network.subnet_ids["postgres-openvpn-subnet"]

  private_service_connection {
    name                           = "${var.environment}-openvpn-sqlprivatelink"
    is_manual_connection           = "false"
    private_connection_resource_id = azurerm_postgresql_server.main.id
    subresource_names              = ["postgresqlServer"]
  }
}
```

### Debug Output/Panic Output

```shell
Can't share debug output as it contains sensitive information, please describe how to locate in it most interesting parts so I can share those.
```

### Expected Behaviour

TF plan should see no changes to be applied.

### Actual Behaviour

TF plan is detecting changes in private_connection_resource_id:
~ private_service_connection {
            name                           = "staging-openvpn-sqlprivatelink"
          ~ private_connection_resource_id = "/subscriptions/4fa2a329-c99a-4ca9-a2dc-acc894ebf84a/resourceGroups/staging-vault-1-rg/providers/Microsoft.DB**for**PostgreSQL/servers/staging-vault-1-postgresql" -> "/subscriptions/4fa2a329-c99a-4ca9-a2dc-acc894ebf84a/resourceGroups/staging-vault-1-rg/providers/Microsoft.DB**For**PostgreSQL/servers/staging-vault-1-postgresql" # forces replacement
          ~ private_ip_address             = "172.14.1.4" -> (known after apply)
            # (2 unchanged attributes hidden)
        }

### References

There is an existing PR for 2.7.0 - https://github.com/hashicorp/terraform-provider-azurerm/pull/12761

**Comment by sinbai** [comment]
@igor-teresco thanks for opening this issue. Unfortunately, I could not reproduce this issue with the following tf configuration and steps. Could you reproduce with it?

**Terraform and AzureRM Provider Version:**
![image](https://user-images.githubusercontent.com/39109137/177117106-7ed14805-9fc0-49e3-8413-54c4c4ea23f1.png)

Step1: Create resource azurerm_private_endpoint with the following tf configuration:
```

terraform {
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "3.7.0"
    }
  }
}

provider "azurerm" {
  features {
  }
}



resource "azurerm_resource_group" "test" {
  name     = "RG-privatelink"
  location = "eastus"
}

resource "azurerm_virtual_network" "test" {
  name                = "vnet-example"
  resource_group_name = azurerm_resource_group.test.name
  location            = azurerm_resource_group.test.location
  address_space       = ["10.5.0.0/16"]
}

resource "azurerm_subnet" "service" {
  name                 = "subnet-example"
  resource_group_name  = azurerm_resource_group.test.name
  virtual_network_name = azurerm_virtual_network.test.name
  address_prefixes     = ["10.5.1.0/24"]

  enforce_private_link_service_network_policies = true
}

resource "azurerm_subnet" "endpoint" {
  name                 = "endpoint-example"
  resource_group_name  = azurerm_resource_group.test.name
  virtual_network_name = azurerm_virtual_network.test.name
  address_prefixes     = ["10.5.2.0/24"]

  enforce_private_link_endpoint_network_policies = true
}

resource "azurerm_postgresql_server" "test" {
  name                = "pe-server-example"
  location            = azurerm_resource_group.test.location
  resource_group_name = azurerm_resource_group.test.name

  sku_name = "GP_Gen5_4"

  storage_mb                   = 5120
  backup_retention_days        = 7
  geo_redundant_backup_enabled = false
  auto_grow_enabled            = true

  administrator_login          = "psqladmin"
  administrator_login_password = "H@Sh1CoR3!"
  version                      = "9.5"
  ssl_enforcement_enabled      = true
}

resource "azurerm_private_dns_zone" "finance" {
  name                = "privatelink.postgres.database.azure.com"
  resource_group_name = azurerm_resource_group.test.name
}

resource "azurerm_private_endpoint" "test" {
  name                = "privatelink-example"
  location            = azurerm_resource_group.test.location
  resource_group_name = azurerm_resource_group.test.name
  subnet_id           = azurerm_subnet.endpoint.id

  private_dns_zone_group {
    name                 = "acctest-dzg-001"
    private_dns_zone_ids = [azurerm_private_dns_zone.finance.id]
  }

  private_service_connection {
    name                           = "acctest-privatelink-psc-001"
    private_connection_resource_id = azurerm_postgresql_server.test.id
    subresource_names              = ["postgresqlServer"]
    is_manual_connection           = false
  }
}

output "id" {
   value = azurerm_postgresql_server.test.id
}
```

Step2: Run `terraform plan` => everything works smoothly. 

**Actual**: There is no changes to be applied about `private_connection_resource_id` on my side. The value of id is "/subscriptions/{subscription id}/resourceGroups/RG-privatelink/providers/Microsoft.DBforPostgreSQL/servers/pe-server-example". `Microsoft.DBforPostgreSQL` instead of `Microsoft.DBForPostgreSQL` was returned by `azurerm_postgresql_server.test.id`.

**Comment by myc2h6o** [comment]
This shall have been fixed by #20452 which auto re-casing `Microsoft.DBforPostgreSQL` in the ID.

**Comment by tombuildsstuff** [comment]
Closing since per the above comment this has been fixed, if this is still an issue in the latest version of the Provider then please let us know and we can take another look.

Thanks!
